dcs/triggers.py

```diff
@@ -48,15 +50,20 @@ def load_from_dict(self, data):
                 mapping.Point(imp_zone["x"], imp_zone["y"]),
                 imp_zone["radius"],
                 imp_zone["hidden"],
-                imp_zone["name"]
+                imp_zone["name"],
+                imp_zone["color"],
+                imp_zone.get("properties",{})
             )
-            tz.color = imp_zone["color"]
             self._zones.append(tz)
             self.current_zone_id = max(self.current_zone_id, tz.id)
 
-    def add_triggerzone(self, position: mapping.Point, radius=1500, hidden=False, name="") -> TriggerZone:
+    def add_triggerzone(self, position: mapping.Point, radius=1500, hidden=False, name="", color=None, properties=None) -> TriggerZone:
         self.current_zone_id += 1
-        tz = TriggerZone(self.current_zone_id, position, radius, hidden, name)
+
+        real_color = color if color is not None else {1: 1, 2: 1, 3: 1, 4: 0.15}
```

Comment:
sorry kinda my fault here, we actually don't need defaults at all.
just pass the parameters to the TriggerZone constructor
defaults will be set anyway in the constructor than if they are None:
```
tz = TriggerZone(self.current_zone_id, position, radius, hidden, name, color, properties)
```

Reply:
I receive `SyntaxError: non-default argument follows default argument` when removing the defaults from `color` and `properties`.  `hidden` and `name` are the culprits here.

Reply:
This works for me?
```
    def add_triggerzone(
            self,
            position: mapping.Point,
            radius=1500, hidden=False,
            name="",
            color=None,
            properties=None
            ) -> TriggerZone:

        self.current_zone_id += 1

        tz = TriggerZone(self.current_zone_id, position, radius, hidden, name, color, properties)
        self._zones.append(tz)
        return tz
```

Reply:
Oh I misunderstood what you were saying.  I thought you were talking about the TriggerZone constructor itself.  I just pushed a commit to mimic what you have above.